### DISCUSSION TITLE
Can't customize QSelect

### DISCUSSION THREAD

**Discussion by npokr** [quasarframework/quasar#11846] [General - Components / Directives / etc] [answered]
I'm trying to write simplest QSelect wrapper with filtering function.
https://codepen.io/npokr/pen/OJxQELm
But the component does not work - the filtering function is called, my options are recalculated, but the options of the internal QSelect are not updated.
The same code works in the template but doesn't work in the wrapper.
What am I doing wrong?

**Comment by metalsadman** [comment, marked as answer]
introduce a new event for updating your parent options so that it's in synced, also you have provided a nonreactive options `stringOptions` on your wrapper.
https://codepen.io/metalsadman/pen/ExwedOE?editors=1010

**Reply by npokr** [reply]
I roughly understand why your code works. But I still don't understand why the official example code does not work in wrapper.

**Reply by metalsadman** [reply]
> I roughly understand why your code works. But I still don't understand why the official example code does not work in wrapper.

i already described it above, furthermore, you can compare your old pen to my fork and update to it to see why yours wasn't working.

**Reply by npokr** [reply]
It came as a surprise to me that attrs cannot be overridden. As soon as I realized it, everything became simple.

**Comment by npokr** [comment]
I managed to build a working wrapper with filtering: https://codepen.io/npokr/pen/qBPJrjZ?editors=101.
It turned out that if options are specified in the wrapper, they can no longer be redefined, the inner QSelect will only use wrapper options. So introducing a new property for non-filtered options solved the problem.
But the described behavior is very unexpected for me. Is it documented somewhere?

**Reply by metalsadman** [reply]
> I managed to build a working wrapper with filtering: https://codepen.io/npokr/pen/qBPJrjZ?editors=101. It turned out that if options are specified in the wrapper, they can no longer be redefined, the inner QSelect will only use wrapper options. So introducing a new property for non-filtered options solved the problem. But the described behavior is very unexpected for me. Is it documented somewhere?

they can be redefined like what I did in my codepen. I don't think it's an unexpected behaviour, what's lacking in the component itself is an event for when the options model updates, hence why I made (`update:options`) an event for it in the wrapper so the parent is aware of the options array update and react to it, you can do `:options @update:options` or just `v-model:options` like what you see in the pen. https://v3.vuejs.org/guide/migration/v-model.html#_3-x-syntax

**Reply by metalsadman** [reply]
for completeness you'll have to forward the slots and account for when you are not using simple arrays for your options, usage of `option-label` etc.. when options are array of objects not containing `label` and `value` props.

**Reply by npokr** [reply]
> hence why I made (update:options) an event for it in the wrapper so the parent is aware of the options array update and react to it

This 'update:options' event is not applicable in my case.  In my case, options are populated with an asynchronous call, so they cannot be remembered in setup function.
But thanks for this trick, I will remember it and apply it in some suitable case.

> for completeness you'll have to forward the slots and account for when you are not using simple arrays for your options, usage of option-label etc.. when options are array of objects not containing label and value props.

Of course, this codepen is a highly simplified example. I know getOptionLabel method.
